tests/TestCase.php

```diff
@@ -46,13 +46,77 @@ abstract class TestCase extends PHPTestCase
 
     protected $rootDir;
 
+    private $catchAllErrorHandler;
+
+    function __construct() {
+        parent::__construct();
+
+        // PHP does not implement setting a class property to an anonymous function,
+        // so we have to do it in the constructor.
+        $this->catchAllErrorHandler = function ($typeNumber, $message, $file, $lineNumber) {
+            $this->fail(
+                sprintf('%s: "%s" in %s:%d',
+                    $this->getErrorType($typeNumber),
+                    $message,
+                    $file,
+                    $lineNumber
+                )
+            );
+        };
+    }
+
     public function setUp()
     {
         parent::setUp();
 
         $this->rootDir = __DIR__.'/..';
     }
 
+    public function tearDown() {
+        // if we changed the error handler using catchAllErrors(), reset it now
+        $currentErrorHandler = set_error_handler('var_dump');
```

Comment:
Using var_dump as error handler seems odd here.

Reply:
This is just a workaround because PHP has no function to get the current error handler. (see https://stackoverflow.com/questions/12378644/how-can-i-retrieve-the-current-error-handler) `var_dump` just acts as a dummy here and is immediately reset in the following line via `restore_error_handler()`. PHPUnit itself uses this in its `DeprecationErrorHandler.php`!

Reply:
OK, will accept it. But can you please paste a link to the code you refer in 

> PHPUnit itself uses this in its DeprecationErrorHandler.php

Reply:
> can you please paste a link into the `tearDown` function to the code you refer in

Sure! To also have this documented here, here it is:
https://github.com/symfony/phpunit-bridge/blob/master/DeprecationErrorHandler.php#L183-L184

Reply:
Thank you.